uwblueprint/childrens-aid-society
# Add new caregiver service

## Notion ticket link

[Add new caregiver](https://www.notion.so/uwblueprintexecs/Add-new-caregiver-dc491e2b05c94e93aaf0f56db9415c3b)



## Implementation description
* Added `Caregiver_DTO` and `CreateCaregiver_DTO`
* Added interface for `Caregiver` service
* Added caregiver services function.



## Steps to test
1. Check if all the tests pass by running - `docker exec -it cas_py_backend /bin/bash -c "pip install -e . && pytest"`



## What should reviewers focus on?
* The valid test cases are yet to be written for the `Caregiver` services because of the different relationships between `Caregiver` model and other existing models. To create a dummy `Caregiver` record for the tests, other dummy models that the `Caregiver` model is associated with needs to created too. Well, that could be done with `pytest fixtures` but for some reason, some weird error shows up.


## Checklist
